File: backend/src/main/java/com/keystone/backend/entity/WorkOrder.java
 package com.keystone.backend.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import com.keystone.backend.entity.User;

@Entity
@Table(name = "work_orders")
public class WorkOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String description;

    private String priority;

    private String status;

    private LocalDateTime createdAt;

    private String createdBy;

    @ManyToOne
    @JoinColumn(name = "site_id")
    private Site site;

    @ManyToOne
@JoinColumn(name = "technician_id")
private User technician;


public User getTechnician() {
    return technician;
}

public void setTechnician(User technician) {
    this.technician = technician;
}
    public WorkOrder() {

}

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getPriority() {
        return priority;
    }

    public void setPriority(String priority) {
        this.priority = priority;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public String getCreatedBy() {
        return createdBy;
    }

    public void setCreatedBy(String createdBy) {
        this.createdBy = createdBy;
    }

    public Site getSite() {
        return site;
    }

    public void setSite(Site site) {
        this.site = site;
    }

}

File: frontend/src/types/workOrder.ts
export interface WorkOrder {
  id: number;
  title: string;
  description: string;
  priority: string;
  status: string;
  createdAt?: string;
  createdBy?: string;
  siteId?: number;
  siteName?: string;
  technicianId?: number;
  technicianName?: string;
}

export interface WorkOrderFormValues {
  title: string;
  description: string;
  priority: string;
  status: string;
  siteId: string;
  technicianId: string;
}

export interface WorkOrderPageResponse {
  content: WorkOrder[];
  totalElements: number;
  totalPages: number;
  number: number;
  size: number;
}

export interface TechnicianOption {
  id: number;
  fullName: string;
}
